From fa2344e1a0ffed7e648e36ac72c8871d817e40e0 Mon Sep 17 00:00:00 2001
From: Carlos Rafael Giani <dv@pseudoterminal.org>
Date: Mon, 8 May 2017 01:38:57 +0200
Subject: [PATCH] Make the detection of libunwind and libdw deterministic

Upstream-Status: Pending

Signed-off-by: Carlos Rafael Giani <dv@pseudoterminal.org>
---
 configure.ac | 28 ++++++++++++++++++++--------
 1 file changed, 20 insertions(+), 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index b151208..b7922ac 100644
--- a/configure.ac
+++ b/configure.ac
@@ -821,16 +821,28 @@ fi
 AM_CONDITIONAL(HAVE_GTK, test "x$HAVE_GTK" = "xyes")
 
 dnl libunwind is optionally used by the leaks tracer
-PKG_CHECK_MODULES(UNWIND, libunwind, HAVE_UNWIND=yes, HAVE_UNWIND=no)
-if test "x$HAVE_UNWIND" = "xyes"; then
-  AC_DEFINE(HAVE_UNWIND, 1, [libunwind available])
-fi
+AC_ARG_WITH([unwind],[AS_HELP_STRING([--with-unwind],[use libunwind])],
+            [], [with_unwind=yes])
+AS_IF([test "x$with_unwind" = xyes],
+      [PKG_CHECK_MODULES(UNWIND, libunwind)
+       HAVE_UNWIND=yes
+       AC_DEFINE(HAVE_UNWIND, 1, [libunwind available])
+      ], [
+       HAVE_UNWIND="no"
+      ]
+)
 
 dnl libdw is optionally used to add source lines and numbers to backtraces
-PKG_CHECK_MODULES(DW, libdw, HAVE_DW=yes, HAVE_DW=no)
-if test "x$HAVE_DW" = "xyes"; then
-  AC_DEFINE(HAVE_DW, 1, [libdw available])
-fi
+AC_ARG_WITH([dw],[AS_HELP_STRING([--with-dw],[use libdw])],
+            [], [with_dw=yes])
+AS_IF([test "x$with_dw" = xyes],
+      [PKG_CHECK_MODULES(DW, libdw)
+       HAVE_DW=yes
+       AC_DEFINE(HAVE_DW, 1, [libdw available])
+      ], [
+       HAVE_DW=no
+      ]
+)
 
 dnl Check for backtrace() from libc
 AC_CHECK_FUNC(backtrace, [
-- 
2.7.4

